python/mxnet/rnn/rnn_cell.py

```diff
@@ -913,6 +913,25 @@ def __call__(self, inputs, states):
         output = symbol.elemwise_add(output, inputs, name="%s_plus_residual" % output.name)
         return output, states
 
+    def unroll(self, length, inputs, begin_state=None, layout='NTC', merge_outputs=None):
+        self.reset()
+
+        self.base_cell._modified = False
+        outputs, states = self.base_cell.unroll(length, inputs=inputs, begin_state=begin_state,
+                                                layout=layout, merge_outputs=merge_outputs)
+        self.base_cell._modified = True
+
+        inputs, _ = _normalize_sequence(length, inputs, layout, merge_outputs)
```

Comment:
The handling of merge_outputs is wrong. If it's none you need to decide based on outputs.